httpcore5/src/main/java/org/apache/hc/core5/http/nio/AsyncPushConsumer.java

```diff
@@ -40,7 +41,7 @@
  */
 public interface AsyncPushConsumer extends AsyncDataConsumer {
 
-    void consumePromise(HttpRequest promise, HttpResponse response, EntityDetails entityDetails) throws HttpException, IOException;
+    void consumePromise(HttpRequest promise, HttpResponse response, EntityDetails entityDetails, HttpContext context) throws HttpException, IOException;
```

Comment:
Note this one definitely is a public API change. Is changing the interface like this acceptable?

Reply:
@ato HC 5.0 is still BETA. We are at liberty to make small changes if it helps keep APIs clean